examples/tour.jl

```diff
@@ -3,192 +3,392 @@
-# facilitate, so let's explore them before we dig deeper into Legolas.jl's other table-centric utilities.
-
-# Declare a `Legolas.Schema` with name `"my-schema"` at version `1` whose *required fields*
-# `a`, `b`, `c`, `d`, and `e` are specified via the provided assignment expressions, then return
-# a corresponding row type that matches the declared schema:
-const MyRow = @row("my-schema@1",
-                   a::Real = sin(a),
-                   b::String = string(a, b, c),
-                   c = [a, b, c],
-                   d::Int,
-                   e)
-
-# `MyRow` is thus an alias for the type returned by the `@row` macro:
-@test MyRow == Legolas.Row{typeof(Schema("my-schema@1"))}
-
-# The `MyRow` type has several useful constructors. Let's start with the constructor that
-# accepts all required fields as keyword arguments:
-row = MyRow(a=1.5, b="hello", c="goodbye", d=2, e=["anything"])
-
-# By examining `row`'s fields, we can see how the assignment expressions from `MyRow`'s
-# `@row` declaration were applied in a simple linear fashion to the input arguments:
-@test row.a == sin(1.5)
-@test row.b == string(sin(1.5), "hello", "goodbye")
-@test row.c == [sin(1.5), string(sin(1.5), "hello", "goodbye"), "goodbye"]
-@test row.d == 2
-@test row.e == ["anything"]
-
-# In fact, the field assignment expressions provided to the `@row` macro are interpolated
-# nearly as-is into the underlying code generated by `@row`. For example, the relevant code
-# generated by `MyRow`'s `@row` declaration looks roughly like:
+# - ...contain the associated schema version's required fields in any order
+# - ...elide required fields, in which case the constructor will assume them to be `missing`
+# - ...contain any other fields in addition to the required fields; such additional fields are simply ignored
 #
-#     function Legolas._transform(::typeof(Legolas.Schema("my-schema", 1));
-#                                 a=missing, b=missing, c=missing, d=missing, e=missing,
-#                                 other...)
-#         a::Real = sin(a)
-#         b::String = string(a, b, c)
-#         c::Any = [a, b, c]
-#         d::Int = d
-#         e::Any = e
-#         return (; a, b, c, d, e, other...)
-#     end
+# Demonstrating a few of these properties:
+
+# Providing the additional non-required field `x` in the input, which is simply ignored:
+fields_with_x = (; fields..., x="x")
+@test NamedTuple(FooV1(fields_with_x)) == fields
+
+# Eliding the required field `c`, which is assigned `missing` in the output:
+foo = FooV1(; a=1.0, b="hi", d=[1, 2, 3])
+@test isequal(NamedTuple(foo), (a=1.0, b="hi", c=missing, d=[1, 2, 3]))
+
+# Providing the non-compliantly-typed field `d::Int`, inducing a `MethodError`:
+@test_throws MethodError FooV1(; a=1.0, b="hi", d=2)
+
+# Implicitly providing the non-compliantly-typed field `d::Missing`, inducing a `MethodError`:
+@test_throws MethodError FooV1(; a=1.0, b="hi")
```

Comment:
yeah, this is basically https://github.com/beacon-biosignals/Legolas.jl/issues/44

I'll give it a small ponder to see if this can be fixed easily in this iteration of the PR. if not, though, i think i'll leave tackling that issue to a subsequent release

Reply:
Okay, ponder has been ponder'd. 

The path to implementing this is a bit intertwined with [this line of thinking](https://github.com/beacon-biosignals/Legolas.jl/pull/54#discussion_r1005849968), funnily enough. Basically, we'd only want to perform the kind of validation that gives us the nice error message *after* construction itself has been attempted, since the inputs are not necessarily expected to de facto be valid outputs. But it's the construction bit that throws the poor error in the first place.

this implies the route to surfacing a better error message is to generate a `try/catch` like this:

```jl
# inner constructor
function FooV1(; x=missing, y=missing)
    try
       x::X = ...
       y::Y = ...
    catch err
        @warn "<better descriptive message, possibly using find_violation>"
        rethrow(err)
    end
    return new(x, y)
end
```

not too bad, but this is just enough extra work that i'd rather tackle #44 in a future PR. will xref this from #44